modules/mcuboot/Kconfig

```diff
@@ -44,6 +44,39 @@ config MCUBOOT_IMAGE_VERSION
 	  valuable for debugging purposes. Format: maj.min.rev+build with
 	  latter parts optional.
 
+config MCUBOOT_HARDWARE_DOWNGRADE_PREVENTION
+	bool "Downgrade prevention using hardware security counters"
+	help
+	  MCUBOOT_HW_DOWNGRADE_PREVENTION Enable downgrade prevention in MCUBOOT using hardware
+	  security counters. When this option is enabled MCUBOOT
+	  will compare the security counter of the image with a hardware
+	  backed security counter and it only allow updating the image
+	  when the security counter is greater or equal to the current value.
+
+config MCUBOOT_HW_DOWNGRADE_PREVENTION_COUNTER_SLOTS
+	int "Number of available hardware counter slots"
+	depends on MCUBOOT_HARDWARE_DOWNGRADE_PREVENTION
+	default 20
+	range 1 200
+	help
+	  When MCUBOOT_HW_DOWNGRADE_PREVENTION is enabled MCUBOOT will use
+	  one hardware counter for each updateable image (UPDATEABLE_IMAGE_NUMBER).
+	  This configuration specifies how many counter slots will be occupied
+	  for the MCUBOOT counters. The hardware counters are stored in OTP storage.
+	  Note that the firmware version is not directly linked to the hardware
+	  counters and thus reducing this number does not limit the number of
+	  firmware upgrades supported by the system.
```

Comment:
No, that means that the hardware rollback protection depends only on the security counter and not on the firmware version. So you can technically do an "infinite" number of firmware upgrades by updating the version of the software and use the same security counter. This will allow for "non security" critical updates to be more often without occupying hardware counter slots.